File: pset1/greedy.c
#include<stdio.h>
#include<math.h>
#include<cs50.h>
int get_n(int);
int main(void)
{
    float dol,rnd;
    int cents;
    
    do
    {
        printf("\nO hai! How much change is owed?\n");
        dol=GetFloat();
    }while(dol<0);
    
    dol=dol*100;
    rnd=roundf(dol);
    cents=(int)rnd;
    
    int n,q,count=0;
    
    while(cents!=0)
    {
        n=get_n(cents);
        q=cents/n;
        cents=cents%n;
        count=count+q;
    }
    
    printf("%i\n",count);
    
    return 0;
}

int get_n(int x)
{
    if(x>=25)
        return 25;
    else if(x>=10)
            return 10;
        else if(x>=5)
                return 5;
            else
                return 1;
}

File: pset6/greedy.py
import cs50

def main():
    
    noofcoins = 0;
    change = get_valid_change()
    change = change * 100
    change = round(change)
    
    while change != 0 :
        cent = largest_cent(change)
        noofcoins = noofcoins + (change // cent)
        change = change % cent
    
    print(noofcoins)

#returns valid change value
def get_valid_change():
    print("O hai! ", end = "")
    while True:
        print("How much change is owed?") 
        n = cs50.get_float()
        if n > 0:
            break
    return n

#returns largest cent value 
def largest_cent(change) :
    if change >= 25 :
        return 25;
    elif change < 25 and change >= 10 :
        return 10;
    elif change < 10 and change >= 5 :
        return 5;
    else:
        return 1;

if __name__ == "__main__":
    main()